### ISSUE TITLE
RStudio ignoring LD_LIBRARY_PATH

### ISSUE THREAD

**Issue by jkh1** [rstudio/rstudio#13100]
### System details

    RStudio Edition : Desktop
    RStudio Version : 2023.03.0 
    OS Version      : Ubuntu 20.04
    R Version       :  4.3.0

### Steps to reproduce the problem
Upgrade RStudio desktop from 2022.07.2 to a newer version

### Describe the problem in detail
RStudio desktop 2023.03.0 (and 2022.12.0) ignore LD_LIBRARY_PATH from the current environment and seem to use preset values.
Starting R in a terminal outside RStudio shows the LD_LIBRARY path defined in the current environment.
Starting RStudio 2023.03.0 from the same terminal results in "/usr/lib/R/lib:/usr/lib/x86_64-linux-gnu:/usr/lib/jvm/default-java/lib/server" no matter what is set in the terminal environment. 
This looks specific to LD_LIBRARY_PATH as other environment variables appear unaffected.
This was fine in RStudio 2022.07.2 and indeed downgrading RStudio to 2022.07.2 fixes this issue.

This looks different from the similarly named issue #9003 because there Sys.getenv('LD_LIBRARY_PATH') would return the correct value although it would be ignored whereas here Sys.getenv('LD_LIBRARY_PATH') return a generic default value and not the one set in the environment.
This looks more similar to issue #13085 except this is not macOS and I am not aware of any Linux equivalent of macOS SIP that could cause this issue.

### Describe the behavior you expected
R behaviour in RStudio should  be consistent with that of R run from a terminal and in particular RStudio should respect pre-existing LD_LIBRARY_PATH content.

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by kevinushey** [comment]
I don't think so; #13085 is specific to macOS so this is likely a separate issue.

**Comment by jkh1** [comment]
I just installed the latest version (2023.03.1) and the issue is gone. Thanks. 
One last comment:  I don't see it mentioned in the release notes so I don't know if it was specifically addressed or if there's a chance of it resurfacing later.

**Comment by Chas-RST-2022** [comment]
Adding this ticket here, https://rstudioide.zendesk.com/agent/tickets/89578 seems to be related.

**Summary:**

The RANN.so library is attempting to load libstdc++.so.6 from /var/lib64 since we can see that the version installed there does not have GLIBCXX_3.4.29, which is why compiling R with GCC 8.5.0 resolves things as a workaround.

- Presumably, /usr/apps/general/spack/sw/linux-rhel8-zen/gcc-8.5.0/gcc-11.3.0-cwx43q6qt46zl5olgckurx67xtg4nuyd/lib64 is installed on their LD_LIBRARY_PATH in their CLI environment

- When a session is started, it would be expected that value would also be included as it is a part of the user's environment.

- We are seeing from their output that /usr/apps/general/spack/sw/linux-rhel8-zen/gcc-8.5.0/gcc-11.3.0-cwx43q6qt46zl5olgckurx67xtg4nuyd/lib64 is no longer present in LD_LIBRARY_PATH, which is why we are defaulting to loading the system default gcc, which is an incompatible version.

**Comment by kevinushey** [comment]
I believe this is resolved in the latest release / latest daily builds of RStudio. If I run RStudio with:

```
LD_LIBRARY_PATH=example /usr/bin/rstudio 
```

then I see:

```
> Sys.getenv("LD_LIBRARY_PATH")
[1] "/usr/lib/R/lib:/usr/lib/aarch64-linux-gnu:/usr/lib/jvm/default-java/lib/server:example"
```

**Comment by jgutman** [comment]
To be tested as part of #12585
